src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -275,6 +275,7 @@ sockinfo_tcp::sockinfo_tcp(int fd):
 	}
 
 	si_tcp_logdbg("TCP PCB FLAGS: 0x%x", m_pcb.flags);
+	g_p_agent->register_cb((agent_cb_t)&sockinfo_tcp::put_agent_msg, (void *)this);
```

Comment:
No need for listen socket...

Reply:
there is no chance to understand how socket is used during creation.

Reply:
You can call register_cb under connect and accept...